PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -77,13 +77,13 @@ public function __construct(
         $host,
         $port,
         $connection_timeout,
-        $read_write_timeout,
+        $read_write_timeout = 130.0,
         $context = null,
         $keepalive = false,
-        $heartbeat = 0
+        $heartbeat = 60
     ) {
-        if ($heartbeat !== 0 && ($read_write_timeout < ($heartbeat * 2))) {
-            throw new \InvalidArgumentException('read_write_timeout must be at least 2x the heartbeat');
+        if ($heartbeat !== 0 && ($read_write_timeout <= ($heartbeat * 2))) {
+            throw new \InvalidArgumentException('read_write_timeout must be greater than 2x the heartbeat');
```

Comment:
This will cause breaking changes to anyone running exactly double of the heartbead :/

Reply:
The change was made to avoid a scenario where the heartbeat arrives as the timeout expires. We had beta and RC builds available _for weeks_ for people to test in their environments. We will be releasing `2.9.1` shortly if you think this is serious enough to revert. The exception is thrown early enough to be noticed easily.

Reply:
@lukebakken I am having to make configuration changes to a production system to stop an exception being thrown before our consumers start. This *is* a breaking change. Please conform to semver.org, this is not backwards compatible if we're having to make changes to get this to work as expected.

Reply:
@ramunasd 

We'll revert this in `2.9.1`. If @andrefigueira or @olivertappin have the time to review other `2.9.0` changes that inadvertently cause issues like this, and ideally submit PRs to address them, it would be greatly appreciated. Our goal was to bring the behavior of timeouts and heartbeats in-line with other libraries.

Reply:
@lukebakken Thank you, we'll do some additional testing, I understand that the RC was available for weeks, but unfortunately, not all companies work on the same timescales. Obviously that's not the problem of the repo, but if semver is the be followed this isn't necessarily an issue, though as it's a slight edge case, I can understand why it was missed.

Reply:
Looking through the rest of the changes, they seem fine. I don't there's an issue here with the changes themselves, the problem is how this has been tagged. We just want this confirmed properly to [semver.org](https://semver.org/)

Reply:
@olivertappin yes, that is the goal. I'd like to remind you that this library is maintained by volunteers, mostly on their spare time, and that you benefit _without cost_. In the future, please watch this repository for preview releases that you can test and therefore contribute back to this project. Preview releases are also announced on the [`rabbitmq-users`](https://groups.google.com/forum/#!forum/rabbitmq-users) mailing list.